src/actions/index.js

```diff
@@ -34,17 +34,26 @@ export const LOAD_DEPENDENCY_INFO_FROM_DISK = 'LOAD_DEPENDENCY_INFO_FROM_DISK';
 export const ADD_DEPENDENCY_START = 'ADD_DEPENDENCY_START';
 export const ADD_DEPENDENCY_ERROR = 'ADD_DEPENDENCY_ERROR';
 export const ADD_DEPENDENCY_FINISH = 'ADD_DEPENDENCY_FINISH';
+export const ADD_DEPENDENCIES_START = 'ADD_DEPENDENCIES_START';
+export const ADD_DEPENDENCIES_ERROR = 'ADD_DEPENDENCIES_ERROR';
+export const ADD_DEPENDENCIES_FINISH = 'ADD_DEPENDENCIES_FINISH';
```

Comment:
Agreed, it is very confusing. So the new flow would roughly look like:

```
button event actions -
ADD_DEPENDENCY (button pressed to add new dependency)
UPDATE_DEPENDENCY (button pressed to update dependency)
DELETE_DEPENDENCY (button pressed to delete dependency)

cli event actions -
INSTALL_DEPENDENCIES_START (begin `npm install` command for given dependenc(ies))
INSTALL_DEPENDENCIES_ERROR
INSTALL_DEPENDENCIES_FINISH
UNINSTALL_DEPENDENCIES_START (begin `npm remove` command for given dependenc(ies))
UNINSTALL_DEPENDENCIES_ERROR
UNINSTALL_DEPENDENCIES_FINISH
UPDATE_QUEUE_WITH_DEPENDENCY (add dependency to relevant queue item)
```

Structuring it like this means that the queue should probably be moved into its own reducer. The only non-intuitive result of this flow is that (un)installing a single dependency would have the following flow:

```
ADD_DEPENDENCY
INSTALL_DEPENDENCIES_START
INSTALL_DEPENDENCIES_FINISH
```

It's just a little odd to have a single dependency install use the plural action name, but I think it's still correct in this case because the plural action would have an array of 1 or more dependencies attached to it.

Reply:
Maybe we can be even more explicit, and rename `ADD_DEPENDENCY` to `CLICK_ADD_DEPENDENCY_BUTTON`. That way it's clear that the user clicking a button is distinct from any subsequent installs/uninstalls.

The only potentially weird thing about that is if we have multiple ways to trigger installing dependencies... but right now there's only 1 button, and no plans to add additional triggers, so I'm not worried :)